mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1913,6 +1912,128 @@ def _apply_order_by_search_logged_models(
 
         return models.order_by(*order_by_clauses)
 
+    def _apply_filter_string_search_logged_models(
+        self,
+        models: sqlalchemy.orm.Query,
+        session: sqlalchemy.orm.Session,
+        experiment_ids: list[str],
+        filter_string: Optional[str],
+    ):
+        import sqlparse
+
+        if not filter_string:
+            return session.query(SqlLoggedModel).filter(
+                SqlLoggedModel.experiment_id.in_(experiment_ids)
+            )
+
+        try:
+            parsed = sqlparse.parse(filter_string)
+        except Exception as e:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid filter string: {filter_string}. {e!r}"
+            ) from e
+
+        if len(parsed) != 1:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid filter string: {filter_string}. Expected a single SQL expression.",
+            )
+
+        class Entity(Enum):
+            ATTRIBUTE = "attributes"
+            METRIC = "metrics"
+            PARAM = "params"
+            TAG = "tags"
+
+            def from_str(s: str) -> "Entity":
+                if s == "attributes":
+                    return Entity.ATTRIBUTE
+                if s == "metrics":
+                    return Entity.METRIC
+                if s == "params":
+                    return Entity.PARAM
+                if s == "tags":
+                    return Entity.TAG
+
+                raise MlflowException.invalid_parameter_value(
+                    f"Invalid entity: {s}. Expected one of {[e.value for e in Entity]}."
+                )
+
+        @dataclass
+        class Comparison:
+            entity: Entity
+            key: str
+            op: str
+            value: str
+
+        comparisons: list[Comparison] = []
+        for stmt in parsed[0]:
+            if isinstance(stmt, sqlparse.sql.Comparison):
+                non_whitespace_tokens = [str(t) for t in stmt.tokens if not t.is_whitespace]
+                if len(non_whitespace_tokens) != 3:
+                    raise MlflowException.invalid_parameter_value(
+                        f"Invalid comparison: {stmt}. Expected a comparison with 3 tokens."
+                    )
+                identifier, op, value = non_whitespace_tokens
+                value = value.strip("'").strip('"')
+                if m := re.match(r"^([a-z]+)\.(.+)", identifier):
+                    comp = Comparison(
+                        entity=Entity.from_str(m.group(1)),
```

Comment:
What if "attributes.xxx" is used? We might not want to allow it

Reply:
`attributes.xxx` should be supported.

Reply:
Emmm... that's weird, they should just use the attribute name directly